Add actor:move-related-descriptions, refs #13189

jraddaoui opened the pull request:
Add a new task to move the related description from one actor to
another. Including all events and subject access point relations.
Include a `skip-index` option to avoid updating the ES index.

djjuhasz commented on lib/task/actor/actorMoveRelatedDescriptionsTask.class.php:
```diff
@@ -0,0 +1,121 @@
+    ));
+
+    $this->namespace = 'actor';
+    $this->name = 'move-related-descriptions';
```
To me `actor:move-related-descriptions` implies moving descriptions in the descriptive hierarchy.  I would prefer `actor:update-description-relations` or `io:update-actor-relations` or something similar.

jraddaoui replied:
Thanks @djjuhasz!

Since we're using the `source` and `target` parameters, I think "move" is better than "update" but I agree that "description relations" is less confusing, so I went with `move-description-relations` in the end. I also struggle in between "relations" and "relationships" so, please, let me know if you think the later is better.

I'd like to hear @fiver-watson thoughts about this too ...

djjuhasz commented on lib/task/actor/actorMoveRelatedDescriptionsTask.class.php:
```diff
@@ -0,0 +1,121 @@
+    $params = array(':targetId' => $target->id, ':sourceId' => $source->id, ':typeId' => QubitTerm::NAME_ACCESS_POINT_ID);
+    $updatedCount += QubitPdo::modify($sql, $params);
+
+    $this->log($updatedCount.' related descriptions moved.');
```
Again, to me "related descriptions moved." implies the descriptions have been moved in the descriptive hierarchy, which is not the case.  I'd prefer language like "[count] description-actor relationships updated".

djjuhasz commented on lib/task/actor/actorMoveRelatedDescriptionsTask.class.php:
```diff
@@ -0,0 +1,121 @@
+    }
+    else
+    {
+      $this->log('The Elasticsearch index has not been updated.');
```
I suggest adding text like "The Elasticsearch index has not been updated. Please run search:populate manually to update the description-actor relationships in the search index."

djjuhasz commented on lib/task/actor/actorMoveRelatedDescriptionsTask.class.php:
```diff
@@ -0,0 +1,121 @@
+      WHERE relation.object_id=:sourceId
+      AND relation.type_id=:typeId
+      AND object.class_name='QubitInformationObject'";
+    $params = array(':targetId' => $target->id, ':sourceId' => $source->id, ':typeId' => QubitTerm::NAME_ACCESS_POINT_ID);
```
This raises a question for me about line lengths and wrapping.  I've been trying to wrap my code lines according to the rules in https://www.php-fig.org/psr/psr-12/ - e.g. aiming for 80 chars per line or less, but allowing 120 chars for legibility, and avoiding >120 chars per line except in extreme cases.   I don't see any recommendations in The [AtoM Coding Standard](https://wiki.accesstomemory.org/wiki/Development/Coding_standard) or [Symfony Coding Standards](https://symfony.com/doc/current/contributing/code/standards.html) for line length, but do we have any consensus among the AtoM devs on line lengths?

jraddaoui replied:
I like that approach @djjuhasz! And I'd love to see a CI process in AtoM that enforces those rules at some point ...

https://github.com/FriendsOfPHP/PHP-CS-Fixer
https://github.com/squizlabs/PHP_CodeSniffer

djjuhasz replied:
@jraddaoui I would also like to add a linter or formatter to our development workflow at some point, but doing so will mean changing some of our current coding standards I think.  I'll make a task for myself to add the line length standard to our coding standard doc as a first step.

jraddaoui replied:
I'm not sure how much can be configured with those tools, but we may be able to reflect our current coding standards in them. Although, I don't think it's a bad idea to use an opinionated code formatter, even if we have to change our style.

djjuhasz reviewed (commented):
Code looks good! I've made a few comments about the language around "moving" related descriptions, and had one question about line lengths, but nothing that is a blocker to merging.

djjuhasz reviewed (approved):
LGTM. 👍

jraddaoui committed:
Add actor:move-description-relations, refs #13189

Add a new task to move the description relations from one actor to
another. Including all events and subject access point relations.
Include a skip-index option to avoid updating the ES index.
